### PR TITLE
New pulp_sync plugin

### PR DESCRIPTION
**Pull request by twaugh**
This is for transferring repository content from a Docker V2 Registry to a Pulp instance.

### CODE REVIEW

**Comment by mmilata** [comment]
LGTM!

**Comment by jpopelka** [comment]
LGTM

File: atomic_reactor/plugins/post_pulp_sync.py [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+# let's silence warnings from dockpulp: there is one warning for every
+# request which may result in tenths of messages: very annoying with
+# "module", it just prints one warning -- this should balance security
+# and UX
+from warnings import filterwarnings
+filterwarnings("module")
+
+
+@contextmanager
+def dockpulp_config(docker_registry, **kwargs):
```
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_pulp_sync.py]: would be awesome to have support for this in dockpulp

File: atomic_reactor/plugins/post_pulp_sync.py [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+        self.workflow.push_conf.add_pulp_registry(self.pulp_registry_name,
+                                                  pulp_registry)
+
+        images = []
+        self.log.info("syncing from docker V2 registry %s",
+                      self.docker_registry)
+        with dockpulp_config(docker_registry=self.docker_registry) as config:
+            for image in self.workflow.tag_conf.primary_images():
+                self.log.info("syncing %s", image.pulp_repo)
+                pulp.syncRepo(config.env, image.pulp_repo,
```
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_pulp_sync.py]: what does this do precisely? will it sync all tags? if so, we need to push image with all tags to distribution

Comment (twaugh) [inline atomic_reactor/plugins/post_pulp_sync.py]: The `tag_and_push` plugin handles that part.

Comment (twaugh) [inline atomic_reactor/plugins/post_pulp_sync.py]: Oh, maybe you mean: 'does sync only copy over newer tags or also delete tags not present in the docker registry?'

Comment (TomasTomecek) [inline atomic_reactor/plugins/post_pulp_sync.py]: hmmm, I hope it won't delete, just update; b/c when we wipe distribution's repos they will contain only a single build, so pulp would remove references to previous builds

File: atomic_reactor/plugins/post_pulp_sync.py (hunk 1 above)
Comment (twaugh) [inline atomic_reactor/plugins/post_pulp_sync.py]: Support for creating a temporary config file?

File: atomic_reactor/plugins/post_pulp_sync.py (hunk 2 above)
Comment (twaugh) [inline atomic_reactor/plugins/post_pulp_sync.py]: I will find out the answer. I'd been under the impression it was more like an 'import'.

Comment (twaugh) [inline atomic_reactor/plugins/post_pulp_sync.py]: Confirmed: it is only additive.

File: atomic_reactor/plugins/post_pulp_sync.py (hunk 1 above)
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_pulp_sync.py]: configuration via API

File: atomic_reactor/plugins/post_pulp_sync.py (hunk 2 above)
Comment (TomasTomecek) [inline atomic_reactor/plugins/post_pulp_sync.py]: cool

**Pull request merged**